app/renderer/abr-document.js

```diff
@@ -319,23 +332,26 @@ AbrDocument.prototype = {
         if (!path) {
             return this.saveAs(callback);
         }
+        // Pause the watcher to avoid triggering callbacks after saving the document
+        this.pauseWatcher();
         var that = this,
-            data = this.getData(),
-            callback2 = function (err) {
-                if (err) {
-                    return dialogs.fileAccessDenied(path, function () {
-                        that.saveAs(callback);
-                    });
-                }
-                that.setClean();
-                that.setPath(path);
-                that.ipcClient.trigger("setWinPath", path);
-                that.updateWindowTitle();
-                if (typeof callback === "function") {
-                    callback();
-                }
-            };
-        return files.writeFile(data, path, callback2);
+            data = this.getData();
+        files.writeFile(data, path, function (err) {
+            if (err) {
```

Comment:
There is a `return` in the `if (err)` block. If this block is reached and the user cancels the "Save As" dialog then the watcher won't be started again.

I would do something like:

```javascript
if (err) {
    that.startWatcher()
    return dialogs.fileAccessDenied(path, function () {
        that.saveAs(callback);
    });
}
```

Or maybe I'm missing something?